docs/preview/yamlgettingstarted-triggers.md

```diff
@@ -0,0 +1,96 @@
+# YAML getting started - YAML triggers (not yet available, for discussion only)
+
+## Simple CI trigger syntax
+
+On push, the YAML file from the branch that was updated, is used to determine whether
+to queue a CI build.
+
+For example, if master is updated, the trigger defined in master will be used. If a different
+branch is updated, the YAML file from that branch will be used to determine whether to queue a
+build.
+
+```yaml
+trigger:
+- master
+- releases/*
```

Comment:
Okay, so clarify here...if I push this to master (release branches don't exist yet), then the master's version of the yaml file will be used upon pushes to master.  If I create a release/1.0 branch, then the release branch's version of the yaml file would be used on push to release/1.0.

What happens if I decide that pushes should happen on..say....dev/foobar?  Can I do this simply by adding an entry for dev/foobar in the dev/foobar branch, or does the 'default' branch need to specify all the tracked branches?

Reply:
I think it makes most sense for the triggers to be the 'union' of all branch's trigger sets, minus the opt out set.

Reply:
The CI triggers will actually be the set that is defined in the version of the file for the particular branch.  So in the case of /refs/heads/dev/foobar a push there would get no build because it is not specified as included.  I could then update the file in the dev/foobar branch to include dev/* and the next push to dev/foobar would start building.  However, if I then took a branch from master before the change from dev/foobar was merged for dev/foobaz a push there would not build as that branch is not yet specified.

So the set of branches is not a union across all branches, it is only the set specified in the version of the file read for the particular commit.